common/src/main/java/com/example/android/uamp/media/UampNotificationManager.kt

```diff
@@ -0,0 +1,116 @@
+package com.example.android.uamp.media
+
+import android.app.PendingIntent
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import android.support.v4.media.session.MediaControllerCompat
+import android.support.v4.media.session.MediaSessionCompat
+import com.google.android.exoplayer2.ExoPlayer
+import com.google.android.exoplayer2.Player
+import com.google.android.exoplayer2.ui.PlayerNotificationManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+const val NOW_PLAYING_CHANNEL = "com.example.android.uamp.media.NOW_PLAYING"
+const val NOW_PLAYING_NOTIFICATION = 0xb339
+
+/**
+ * A wrapper class for ExoPlayer's PlayerNotificationManager. It sets up the notification shown to
+ * the user during audio playback and provides track metadata, such as track title and icon image.
+ */
+class UampNotificationManager(private val context: Context,
+                              private val player: ExoPlayer,
+                              sessionToken: MediaSessionCompat.Token,
+                              notificationListener: PlayerNotificationManager.NotificationListener) {
+
+    private val serviceJob = SupervisorJob()
+    private val serviceScope = CoroutineScope(Dispatchers.Main + serviceJob)
+    private val notificationManager: PlayerNotificationManager
+
+    init {
+        val mediaController = MediaControllerCompat(context, sessionToken)
+
+        notificationManager = PlayerNotificationManager(
+                context,
+                NOW_PLAYING_CHANNEL,
+                NOW_PLAYING_NOTIFICATION,
+                DescriptionAdapter(mediaController),
+                notificationListener).apply {
+
+            setMediaSessionToken(sessionToken)
+            setSmallIcon(R.drawable.ic_notification)
+
+            // Don't display the rewind or fast-forward buttons.
+            setRewindIncrementMs(0)
+            setFastForwardIncrementMs(0)
+
+            // Display the stop buton. This can be used to easily stop playback and dismiss the
+            // notification.
+            setUseStopAction(true)
```

Comment:
I feel like on API 21+ it should be possible to make the notification dismissible, and use that to stop playback, which feels like a more natural motion to me? Displaying the stop button on API 19 works though. Thoughts?

Reply:
I took a look at the way the Spotify, Google Play Music and YouTube handle this. Only YouTube shows a close button (which is subtly different to "stop"): 

![image](https://user-images.githubusercontent.com/873212/83635231-c614fe80-a59b-11ea-9456-ada17c24cba2.png)

My conclusion is that it's better to not have the stop button with the current square icon as it's not immediately obvious that it'll dismiss the notification.

Note: The notification is already dismissible by swiping on API 21+. On API 19 a close button is displayed when playback is paused: 

![image](https://user-images.githubusercontent.com/873212/83635306-e9d84480-a59b-11ea-8dc2-6fbafca4909a.png)